src/util/metrics/threads_linux.rs

```diff
@@ -228,6 +272,67 @@ fn get_thread_stat_internal(tid: pid_t, stat: &str) -> Result<Stat> {
     })
 }
 
+// I/O statistics for threads.
+struct Io {
+    // Attempt to count the number of bytes which this process really did cause
+    // to be fetched from the storage layer.  This is accurate for block-backed
+    // filesystems.
+    read_bytes: u64,
+    // Attempt to count the number of bytes which this process caused to be
+    // sent to the storage layer.
+    write_bytes: u64,
+}
+
+impl Io {
+    // # cat /proc/3828/io
+    // rchar: 323934931
+    // wchar: 323929600
+    // syscr: 632687
+    // syscw: 632675
+    // read_bytes: 0
+    // write_bytes: 323932160
+    // cancelled_write_bytes: 0
+    const READ_BYTES_INDEX: usize = 4;
+    const WRITE_BYTES_INDEX: usize = 5;
+
+    fn collect(pid: pid_t, tid: pid_t) -> Result<Io> {
+        let mut io = String::new();
+        fs::File::open(format!("/proc/{}/task/{}/io", pid, tid))
+            .and_then(|mut f| f.read_to_string(&mut io))?;
+        get_thread_io_internal(&io)
+    }
+}
+
+// Extracted from `Io::collect`, for test purpose.
+fn get_thread_io_internal(io: &str) -> Result<Io> {
+    let read_bytes = io
```

Comment:
please extract a `let lines = io.lines()` to avoid repeat line split. rest LGTM.

Reply:
I tried, and found out it became less straightforward, the link below is an example.
It's not performance critical, so I think this is fine.

https://play.rust-lang.org/?gist=88f63c8d5a8000bb683a26bc2ec68d72&version=stable&mode=debug&edition=2015

```rust
#![allow(unused)]
fn main() {
let a = [1, 2, 3];

let mut iter = a.iter();

assert_eq!(iter.nth(0), Some(&1));
assert_eq!(iter.nth(1), Some(&2));
}

thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Some(3)`,
 right: `Some(2)`', src/main.rs:8:1
```